torch/csrc/jit/script/compiler.cpp

```diff
@@ -1424,25 +1421,34 @@ struct to_ir {
 
   // Desugars slice syntactic sugar tensor[begin:end] -> tensor.slice(begin,
   // end).
-  Value* emitSlice(
-      const SourceRange& loc,
-      TreeList&& inputs) {
-    const auto applyInputs =
-        Compound::create(TK_LIST, loc, std::move(inputs));
-    const auto input_values = getNamedValues(applyInputs->trees(),
-                                             /*maybe_unpack*/false,
-                                             identity);
+  Value* emitSlice(const Slice& slice) {
+    const auto& loc = slice.range();
+    TreeList inputs = {slice.value(), slice.startOr(0)};
+    const auto applyInputs = Compound::create(TK_LIST, loc, std::move(inputs));
+    const auto input_values = getNamedValues(
+        applyInputs->trees(),
+        /*maybe_unpack*/ false,
+        identity);
+
     NamedValue tensor = input_values[0];
     NamedValue begin = input_values[1];
-    NamedValue end = input_values[2];
-    NamedValue dim = NamedValue(loc, "dim",
-        graph->insertConstant(0, loc));
-    NamedValue step = NamedValue(loc, "step",
-        graph->insertConstant(1, loc));
-
-    return emitBuiltinCall(
-               loc, method, "slice", {tensor, dim, begin, end, step}, {}, true)
-        ->asValue(loc, method);
+    NamedValue dim = NamedValue(loc, "dim", graph->insertConstant(0, loc));
+    NamedValue step = NamedValue(loc, "step", graph->insertConstant(1, loc));
+
+    const auto has_end = slice.end().present();
+    if (has_end) {
+      // If the user specified an `end` index, pass it down
```

Comment:
this might be simpler to maintain:

```
std::vector<NamedValue> args = {tensor, dim, begin};
if(has_end)
  args.push_back(end);
return emitBuiltinCall(
                 loc, method, "slice", {tensor, dim, begin}, {step}, true)
          ->asValue(loc, method);
```